nodeadm/internal/aws/imds/imds.go

```diff
@@ -26,8 +26,12 @@ func dynamicProxyFunc(req *http.Request) (*url.URL, error) {
 	// Link-local addresses for IMDS do not need to be going through a proxy
 	// The IMDS has two endpoints on an instance: IPv4 (169.254.169.254) and IPv6 ([fd00:ec2::254])
 	// https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-options.html
-	if req.URL.Host == "169.254.169.254" || req.URL.Host == "[fd00:ec2::254]" {
-		return nil, nil
+	hostname := req.URL.Hostname()
+	bypassHosts := []string{"169.254.169.254", "[fd00:ec2::254]", "localhost"}
+	for _, host := range bypassHosts {
+		if hostname == host {
+			return nil, nil
+		}
 	}
```

Comment:
this could be done with a `slices` stdlib function:
```suggestion
	if slices.Contains(bypassHosts, host)  {
		return nil, nil
	}
```